### DISCUSSION TITLE
Custom Fastfile Location

### DISCUSSION THREAD

**Discussion by Sherlouk** [fastlane/fastlane#20274] [Q&A]
Is it possible to override the location of the Fastfile (or other fastlane files)? It currently uses a folder called `fastlane` but it'd be nice if we could move this to a different directory.

> The Fastfile has to be inside your ./fastlane directory.

Source: https://docs.fastlane.tools/advanced/Fastfile/

Assuming it's not currently possible, but thought I'd ask 😄 

Thanks

**Comment by Sherlouk** [comment]
Looks like this might be the thing which controls the location?

https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/fastlane_folder.rb#L5

Could potentially be enhanced to support being overridden with an environment variable?

**Reply by lacostej** [reply]
Note that a  `Fastfile` can also be located in your current directory (as per the previous answer).

You can also import Fastfile from another folder, so that you would use a minimum locally installed fastfile. See https://docs.fastlane.tools/advanced/Fastfile/

Don't forget that if you plan on moving your fastfile around, you will perhaps have to adjust your Fastfile code to know where you are invoking fastlane from. See **Directory behavior** in https://docs.fastlane.tools/advanced/fastlane/.

If you reason for moving the fastfile around is to share it accross multiple projects, there are other options, such as `import` above or `import_from_git`.

**Reply by Sherlouk** [reply]
I suppose my use case is more of an OCPD need - the want for it to be Fastlane not fastlane 😂

Or perhaps store it under an Automation folder.
